x-pack/filebeat/fbreceiver/receiver_leak_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package fbreceiver
+
+import (
+	"context"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/goleak"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+func TestLeak(t *testing.T) {
+	// opencensus goroutine comes from init in cloud.google.com/go/pubsub and filebeat/input/gcppubsub
+	defer goleak.VerifyNone(t,
+		goleak.IgnoreAnyFunction("github.com/Microsoft/go-winio.getQueuedCompletionStatus"),
+		goleak.IgnoreAnyFunction("go.opencensus.io/stats/view.(*worker).start"))
+
+	monitorSocket := genSocketPath()
+	var monitorHost string
+	if runtime.GOOS == "windows" {
+		monitorHost = "npipe:///" + filepath.Base(monitorSocket)
+	} else {
+		monitorHost = "unix://" + monitorSocket
+	}
+	config := Config{
+		Beatconfig: map[string]any{
+			"filebeat": map[string]any{
+				"inputs": []map[string]any{
+					{
+						"type":    "benchmark",
+						"enabled": true,
+						"message": "test",
+						"count":   1,
+					},
+					{
+						"type":    "filestream",
+						"enabled": true,
+						"id":      "must-be-unique",
+						"paths":   []string{"none"},
+					},
+				},
+			},
+			"output": map[string]any{
+				"otelconsumer": map[string]any{},
+			},
+			"logging": map[string]any{
+				"level": "debug",
+				"selectors": []string{
+					"*",
+				},
+			},
+			"path.home":               t.TempDir(),
+			"http.enabled":            true,
+			"http.host":               monitorHost,
+			"queue.mem.flush.timeout": "0s",
+			"processors": []map[string]any{
+				{
+					"add_host_metadata": map[string]any{
+						"when.not.contains.tags": "forwarded",
+					},
+				},
+			},
+		},
+	}
+
+	factory := NewFactory()
+	var receiverSettings receiver.Settings
+	observedCore, observedLogs := observer.New(zapcore.DebugLevel)
+	receiverSettings.Logger = zap.New(observedCore)
+	receiverSettings.ID = component.NewIDWithName(factory.Type(), "r1")
+
+	var consumeLogs DummyConsumer
+	rec, err := factory.CreateLogs(t.Context(), receiverSettings, &config, &consumeLogs)
+	require.NoError(t, err)
+	require.NoError(t, rec.Start(t.Context(), nil))
+	if !assert.Eventually(t,
```

Comment:
Both of these asserts could easily be a require, but that doesn't need to block merging this PR.